File: backend/src/main/java/com/back/global/security/SecurityUser.java
package com.back.global.security;

import org.jspecify.annotations.Nullable;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

import java.util.Collection;

public class SecurityUser extends User {

    private int id;
    private String nickname;

    public SecurityUser(int id, String username, @Nullable String password, String nickname, Collection<? extends GrantedAuthority> authorities) {
        super(username, password, authorities);
        this.id = id;
        this.nickname = nickname;
    }

    public int getId() {
        return id;
    }

    public String getNickname() {
        return nickname;
    }
}

File: backend/src/main/java/com/back/global/security/SecurityUser.kt
package com.back.global.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

class SecurityUser(
    val id: Int,
    username: String,
    password: String?,
    val nickname: String,
    authorities: MutableCollection<out GrantedAuthority>
) : User(username, password, authorities)
